src/frontend/src/optimizer/plan_node/logical_scan.rs

```diff
@@ -581,12 +581,33 @@ impl LogicalScan {
             return None;
         }
 
+        let output_col_map = self
+            .output_col_idx()
+            .iter()
+            .cloned()
+            .enumerate()
+            .map(|(id, col)| (col, id))
+            .collect::<BTreeMap<_, _>>();
+        let unmatched_idx = output_col_map.len();
         let index = self.indexes().iter().find(|idx| {
+            let s2p_mapping = idx.secondary_to_primary_mapping();
             Order {
                 column_orders: idx
-                    .index_item
+                    .index_table
+                    .pk()
                     .iter()
-                    .map(|idx_item| ColumnOrder::new(idx_item.index, OrderType::ascending()))
+                    .map(|idx_item| {
+                        ColumnOrder::new(
+                            *output_col_map
+                                .get(
+                                    s2p_mapping
+                                        .get(&idx_item.column_index)
+                                        .expect("should be in s2p mapping"),
```

Comment:
I think returning an `internal error` here is better than calling `expect`. In case there's some error, at least the frontend won't crash, can continue serving other queries.

Reply:
It must be existing, otherwise, we have a corrupted index catalog.